tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/signup.cpp

```diff
@@ -183,6 +183,75 @@ tcf_err_t tcf::enclave_api::enclave_data::CreateEnclaveData(
     return result;
 }  // tcf::enclave_api::base::CreateSignupData
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t tcf::enclave_api::enclave_data::RefreshEnclaveKey(
+    const Base64EncodedString& inSealedEnclaveData,
+    StringArray& outPublicEnclaveData,
+    Base64EncodedString& outSealedEnclaveData) {
+    tcf_err_t result = TCF_SUCCESS;
+
+    try {
+        tcf_err_t presult = TCF_SUCCESS;
+        sgx_status_t sresult;
+
+        ByteArray in_sealed_enclave_data = \
+            Base64EncodedStringToByteArray(inSealedEnclaveData);
+
+        outPublicEnclaveData.resize(CalculatePublicEnclaveDataSize());
+
+        ByteArray sealed_enclave_data_buffer(CalculateSealedEnclaveDataSize());
+
+        // Get the enclave id for passing into the ecall
+        sgx_enclave_id_t enclaveid = g_Enclave[0].GetEnclaveId();
+
+        size_t computed_public_enclave_data_size = 0;
+        size_t computed_sealed_enclave_data_size = 0;
+
+        sresult = g_Enclave[0].CallSgx(
+            [enclaveid,
+             &presult,
+             in_sealed_enclave_data,
+             &outPublicEnclaveData,
+             &computed_public_enclave_data_size,
+             &sealed_enclave_data_buffer,
+             &computed_sealed_enclave_data_size ] () {
+                sgx_status_t ret = ecall_RefreshEnclaveKey(
+                    enclaveid,
+                    &presult,
+                    in_sealed_enclave_data.data(),
+                    in_sealed_enclave_data.size(),
+                    outPublicEnclaveData.data(),
+                    outPublicEnclaveData.size(),
+                    &computed_public_enclave_data_size,
+                    sealed_enclave_data_buffer.data(),
+                    sealed_enclave_data_buffer.size(),
+                    &computed_sealed_enclave_data_size);
+                return tcf::error::ConvertErrorStatus(ret, presult);
+            });
+
+        tcf::error::ThrowSgxError(sresult, "SGX enclave call failed (ecall_RefreshEnclaveKey), failed to refresh enclave encryption key");
```

Comment:
Put the error string on a new line to make it easier to read.

Reply:
Done